test/functional/tests.py

```diff
@@ -1,14 +1,52 @@
 # _*_ coding: utf-8 -*-
 import os
-import shutil
 import json
+import shutil
+import socket
+import sys
 import unittest
+import logging
 
 import ipfsApi
 
 
+__is_available = NotImplemented
+def is_available():
+    """
+    Return whether the IPFS daemon is reachable or not
+    """
+    global __is_available
+    
+    if not isinstance(__is_available, bool):
+        s = socket.socket()
+        try:
+            s.connect((ipfsApi.default_host, ipfsApi.default_port))
+        except IOError:
+            __is_available = False
+        else:
+            __is_available = True
+        finally:
+            s.close()
+    
+    return __is_available
+
+
+def skipIfOffline():
+    if is_available():
+        return lambda func: func
```

Comment:
So I follow that this is a decorator, but not how it works when you're decorating the test classes. Could you explain the interaction to me? How does this work since `skipIfOffline` doesn't accept parameters?